packages/jsActions/nanoflow-actions-native/src/geolocation/GetCurrentLocation.ts

```diff
@@ -38,20 +40,35 @@ export async function GetCurrentLocation(
 
     return new Promise((resolve, reject) => {
         const options = getOptions();
-        navigator.geolocation.getCurrentPosition(onSuccess, onError, options);
+        const watchId: number = navigator.geolocation.watchPosition(onSuccess, onError, options);
+        const timeStart = Date.now();
+        let lastAccruedPosition: GeolocationResponse;
 
-        function onSuccess(position: GeolocationResponse): void {
+        function createGeolocationObject(position: GeolocationResponse): void {
             mx.data.create({
                 entity: "NanoflowCommons.Geolocation",
-                callback: mxObject => {
-                    const geolocation = mapPositionToMxObject(mxObject, position);
-                    resolve(geolocation);
-                },
+                callback: mxObject => resolve(mapPositionToMxObject(mxObject, position)),
                 error: () =>
                     reject(new Error("Could not create 'NanoflowCommons.Geolocation' object to store location"))
             });
         }
 
+        function onSuccess(position: GeolocationResponse): void {
+            if (watchId && (!minimumAccuracy || minimumAccuracy >= position.coords.accuracy)) {
+                navigator.geolocation.clearWatch(watchId);
+                createGeolocationObject(position);
+            } else {
+                if (!lastAccruedPosition || position.coords.accuracy < lastAccruedPosition.coords.accuracy) {
```

Comment:
I'm reading this as "if the new position has a higher accuracy than the last accrued position, then swap it out", shouldn't this be the other way around? Or am I misunderstanding that a higher value of `accuracy` means better quality?

Reply:
Nvm, I misunderstood the docs. `accuracy` it the amount of meters, then this is fine.